src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,650 @@
+            // the ConfigFactory.  Therefore they should be null:
+            if ( addToSystemProperties.get(i) ) {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), null);
+            }
+            // If the filename didn't exist in the system properties, then we did need to add them to the
+            // ConfigFactory properties:
+            else {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), ConfigUtils.NO_PATH_VARIABLE_VALUE);
+            }
+        }
+
+        // Remove these from our System Properties and ConfigFactory properties:
+        for ( int i = 0 ; i < sourcesFilenames.size() ; ++i ) {
+            if ( addToSystemProperties.get(i) ) {
+                System.clearProperty(sourcesFilenames.get(i));
+            }
+            else {
+                ConfigFactory.clearProperty(sourcesFilenames.get(i));
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), null);
+            }
+        }
+    }
+
+    @Test(dataProvider= "createConfigListsAndResults")
+    void testConfigUtilsGetConfigPathVariableNamesFromConfigClasses(final List<Class<?>> configurationClasses, final List<String> expectedFilePathVariables ) {
+        Assert.assertEquals( ConfigUtils.getConfigPathVariableNamesFromConfigClasses(configurationClasses), expectedFilePathVariables );
+    }
+
+    @Test(dataProvider = "createConfigFileAndVariableNames")
+    void testGetSourcesAnnotationPathVariables(final Class<? extends Config> configClass, final List<String> expectedFilePathVariables ) {
+            Assert.assertEquals( ConfigUtils.getSourcesAnnotationPathVariables(configClass), expectedFilePathVariables );
+    }
+
+    @Test(dataProvider = "createConfigAndExpectedProperties")
+    <T extends Config> void testConfigUtilsGetSystemPropertiesFromConfig(final T config, final Map<String, String> expectedProperties) {
+        Assert.assertEquals( ConfigUtils.getSystemPropertiesFromConfig(config), expectedProperties );
+    }
+
+    @Test(dataProvider = "createPropertiesToInjectIntoSystem")
+    void testConfigUtilsInjectToSystemProperties(final Map<String, String> propertiesToInject) {
+
+        // Ensure our properties are not there to start with:
+        for ( final String key : propertiesToInject.keySet() ) {
+            Assert.assertEquals(System.getProperty(key), null);
+        }
+
+        // Inject the stuff:
+        ConfigUtils.injectToSystemProperties(propertiesToInject);
+
+        // Verify our injection:
+        for ( final String key : propertiesToInject.keySet() ) {
+            Assert.assertEquals(System.getProperty(key), propertiesToInject.get(key));
+        }
+
+        // Reset our properties after:
+        for ( final String key : propertiesToInject.keySet() ) {
+            System.clearProperty(key);
+            Assert.assertEquals(System.getProperty(key), null);
+        }
```

Comment:
Could you double-check that all (other) tests in this class that modify system properties reset them afterwards? Do any test cases change the values of system properties that would affect other tests in the test suite (such as the htsjdk settings)?

Reply:
Double checked and everything is being reset.

We don't change any significant values in the unit tests - the only System properties that are actually being injected are the ones from `SystemTestConfig.java`.